openqaoa/workflows/optimizer.py

```diff
@@ -430,42 +536,206 @@ def compile(self, problem: QUBO = None, verbose: bool = True):
             !NotYetImplemented! Set true to have a summary of RQAOA to displayed after compilation
         """
 
-        self.qaoa.compile(problem, verbose=False)
+        # save the original problem
+        self.problem = problem 
+
+        # Create the qaoa object with the properties
+        self._q = QAOA(self.device)
+        self._q.circuit_properties  = self.circuit_properties
+        self._q.backend_properties  = self.backend_properties
+        self._q.classical_optimizer = self.classical_optimizer
 
-    def optimize(self):
+        # compile qaoa object
+        self._q.compile(problem, verbose=verbose)
+
+
+        if self.rqaoa_parameters.rqaoa_type.lower() == "custom":
+
+            n_cutoff = self.rqaoa_parameters.n_cutoff
+            n_qubits = problem.n
+            counter  = self.rqaoa_parameters.counter
+
+            # If schedule for custom RQAOA is not given, we create a schedule such that one spin is eliminated at a time
+            if type(self.rqaoa_parameters.steps) is int:
+                self.rqaoa_parameters.steps = [self.rqaoa_parameters.steps]*(n_qubits-n_cutoff)
+            
+            # Ensure there are enough steps in the schedule
+            assert np.abs(n_qubits - n_cutoff - counter) <= sum(self.rqaoa_parameters.steps),\
+                f"""Schedule is incomplete, add 
+                    {np.abs(n_qubits - n_cutoff - counter) - sum(self.rqaoa_parameters.steps)} 
+                    more eliminations"""
+
+            self.compiled = True
+        
+        elif not self.rqaoa_parameters.rqaoa_type.lower() == "adaptive": 
```

Comment:
I am not sure I get this:

if custom --> do something
elif not adaptive -- > raise an execption?

Is this correct?

Reply:
Yes, so 
if not custom nor adaptative -> raise exception 
if custom -> do something

Reply:
Maybe change the exception clause to `if not self.rqaoa_parameters.rqaoa_type.lower() in [''custom', 'adaptive']`

I don't think the elif is the correct clause for this scenario